src/alloc.c

```diff
@@ -641,9 +640,16 @@ jl_method_t *jl_new_method(jl_lambda_info_t *definition, jl_sym_t *name, jl_tupl
     m->name = name;
     m->sig = sig;
     if (jl_svec_len(tvars) == 1)
-        tvars = (jl_svec_t*)jl_svecref(tvars, 0);
-    m->tvars = tvars;
+        m->tvars = (jl_svec_t*)jl_svecref(tvars, 0);
+    else
+        m->tvars = tvars;
     JL_GC_PUSH1(&m);
+    int i;
+    for(i=(int)jl_svec_len(tvars)-1; i >= 0 ; i--) {
```

Comment:
The spacing seems a little inconsistent with the other `for`s here. Might be better as
```c
for (i = (int)jl_svec_len(tvars)-1; i >= 0; i--)
```
Just a suggestion.